cpp/src/io/parquet/reader_impl_helpers.hpp

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "compact_protocol_reader.hpp"
+#include "parquet_gpu.hpp"
+
+#include <cudf/fixed_point/fixed_point.hpp>
+#include <cudf/io/datasource.hpp>
+#include <cudf/types.hpp>
+
+#include <tuple>
+#include <vector>
+
+namespace cudf::io::detail::parquet {
+
+using namespace cudf::io::parquet;
+
+/**
+ * @brief Function that translates Parquet datatype to cuDF type enum
+ */
+type_id to_type_id(SchemaElement const& schema,
+                   bool strings_to_categorical,
+                   type_id timestamp_type_id);
+
+/**
+ * @brief Converts cuDF type enum to column logical type
+ */
+inline data_type to_data_type(type_id t_id, SchemaElement const& schema)
+{
+  return t_id == type_id::DECIMAL32 || t_id == type_id::DECIMAL64 || t_id == type_id::DECIMAL128
+           ? data_type{t_id, numeric::scale_type{-schema.decimal_scale}}
+           : data_type{t_id};
+}
+
+/**
+ * @brief The row_group_info class
+ */
+struct row_group_info {
+  size_type const index;
+  size_t const start_row;  // TODO source index
+  size_type const source_index;
```

Comment:
Any idea why the TODO is still here?

Also, I don't think these members should be `const`, it only does harm by preventing moves.

Reply:
@nvdbaranec Do you have any idea why TODO?